apps/sms/views/conversation/test/unit/conversation_test.js

```diff
@@ -1585,183 +1585,213 @@ suite('conversation.js >', function() {
       );
     }
 
-    setup(function(done) {
-      fakeMessage = MockMessages.sms({
+    setup(function() {
+      this.sinon.stub(Navigation, 'isCurrentPanel');
+      this.sinon.spy(ConversationView, 'appendMessage');
+    });
+
+
+    test('onMessageSent for MMS', function(done) {
+      fakeMessage = MockMessages.mms({
         id: 24601,
         delivery: 'sending'
       });
 
-      this.sinon.stub(Navigation, 'isCurrentPanel').
-        withArgs('thread', { id: fakeMessage.threadId }).
-        returns(true);
-
-      this.sinon.spy(ConversationView, 'appendMessage');
+      Navigation.isCurrentPanel.withArgs(
+        'thread', { id: fakeMessage.threadId }
+      ).returns(true);
 
       MessageManager.on.withArgs('message-sending').yield({
         message: fakeMessage
       });
 
-      ConversationView.appendMessage.returnValues[0].then(() => {
+      MessageManager.on.withArgs('message-failed-to-send').yield({
+        message: fakeMessage
+      });
+
+      ConversationView.appendMessage.lastCall.returnValue.then(() => {
         container = document.getElementById('message-' + fakeMessage.id);
+        assertMessageStatus('error');
       }).then(done, done);
     });
```

Comment:
Wondering if it's better to assert the this.buildingMessages is added and cleared since we don't want the unexpected binding and cached for DOM element after appended.